hw2/emjames/q2/calcAngle.cpp

```diff
@@ -0,0 +1,67 @@
+// Write a C++ function that calculates the angle (in radians) between two vectors in the 2-dimensional Cartesian coordinate system.  Use `pybind11 to wrap it to Python. Use Python unit-test to check the result is correct.  The test runner should be py.test.
+
+#include <pybind11/pybind11.h>
+#include <iostream>
+#include <vector>
+#include <numeric>
+#include <math.h>
+#include "calcAngle.h"
+
+
+int main(int, char **)
+{
+
+    std::string v1_str = "";
+    std::string v2_str = "";
+
+
+    std::cout << "Enter first vector (i.e: 0,1): ";
+    std::cin >> v1_str;
+
+    std::cout << "Enter second vector (i.e: 1,0): ";
+    std::cin >> v2_str;
+
+    double angle = findAngle2d(v1_str, v2_str);
+    std::cout << "Angle: " << angle << std::endl;
+
+
+    return 0;
+}
+
+
+std::vector<float> str2VecFloat(std::string str)
+{
+
+    int c = str.find(",");
+    float n1 = std::stof(str.substr(0, c));
+    float n2 = std::stof(str.substr(c+1));
+
+    std::vector <float> vec = {n1, n2};
+
+    return vec;
+}
+
+double findAngle2d(std::string v1_str, std::string v2_str)
+{
+    // Convert strings to float
+    std::vector<float> v1 = str2VecFloat(v1_str);
+    std::vector<float> v2 = str2VecFloat(v2_str);
+
+    // 1. Compute a dot product
+    double product = std::inner_product(v1.begin(), v1.end(), v2.begin(), 0);
```

Comment:
I think you need to modify 0 to v2.end()

Reply:
Thanks for the review. 

The parameters of `std::inner_product` are as follows:
`T inner_product( InputIt1 first1, InputIt1 last1, InputIt2 first2, T init );`

**first1, last1** - the first range of elements
**first2** - the beginning of the second range of elements
**init** - initial value of the sum of products

The 0 is the initial sum of the products. 
 [CPP Reference](https://en.cppreference.com/w/cpp/algorithm/inner_product)

Reply:
I see